pkg/cluster/deploybaseresources.go

```diff
@@ -35,6 +37,35 @@ func (m *manager) createDNS(ctx context.Context) error {
 	return m.dns.Create(ctx, m.doc.OpenShiftCluster)
 }
 
+func (m *manager) createOIDC(ctx context.Context) error {
+	blobContainerName := env.OIDCBlobContainerPrefix + m.doc.ID
```

Comment:
Missing condition at top for idempotency.  
```
if m.doc.OpenShiftCluster.Properties.ClusteRProfile.OIDCIssuer != nil {
    return nil
}

Reply:
This may need to change if we support rotating the SA signing key.  I would say out of scope for now, but to consider for future.

Reply:
Since during the cluster installation OIDCIssuer is set as part of the createOIDC function, below check wouldn't allow the OIDCIssuer configuration.
```
if m.doc.OpenShiftCluster.Properties.ClusteRProfile.OIDCIssuer != nil {
    return nil
}
```

The check if the cluster is MIWI was supposed to be performed here:- https://github.com/Azure/ARO-RP/pull/3564/files#diff-71a1dd610bf07a9e4ece9f782af2bad11e6f2d0e03df442019088f8c47265497R283

With a nil check on `m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile`.
I will now make the add the check in the function itself:-

```
if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
    return nil
}
```

Reply:
Uncommented the condition now for making `createOIDC` idempotent.